zerver/tests/test_invite.py

```diff
@@ -997,6 +999,46 @@ def test_multi_user_invite(self) -> None:
             ]
         )
 
+    def test_direct_notification_for_accepted_invitation(self) -> None:
+        """
+        New test to check if notification-bot message is sent to the inviter
+        when send_notification is false.
+        """
+
+        self.login("hamlet")
+        user_profile = self.example_user("hamlet")
+        invitee = self.nonreg_email("alice")
+        realm = get_realm("zulip")
+        self.assert_json_success(self.invite(invitee, ["Denmark"], False))
+        self.assertTrue(find_key_by_email(invitee))
+        self.submit_reg_form_for_user(invitee, "password")
+
+        assert user_profile.recipient_id is not None
+        last_inviter_message = Message.objects.filter(
+            recipient_id=user_profile.recipient_id, realm=realm
+        ).last()
+
+        self.assertIsNone(last_inviter_message, "Unexpected inviter message found.")
+
+        self.login("hamlet")
+        new_invitee = self.nonreg_email("bob")
+        self.assert_json_success(self.invite(new_invitee, ["Denmark"], True))
+        self.assertTrue(find_key_by_email(new_invitee))
+        self.submit_reg_form_for_user(new_invitee, "new_password")
+
+        new_invitee_profile = self.nonreg_user("bob")
+        new_last_inviter_message = Message.objects.filter(
```

Comment:
We use `referrer` to refer to the person who sent the invite, so would be better to be consistent (this can be changed in comment as well above). Or, a better variable name would be `invite_acceptance_notification_message`.